pandas/tests/indexes/test_setops.py

```diff
@@ -799,8 +799,16 @@ def test_difference_empty_arg(self, index, sort):
         first = index[5:20]
         first.name = "name"
         result = first.difference([], sort)
+        expected = first.unique()
```

Comment:
Could you construct a completely new `Index` for `expected instead (and below)?

Reply:
Updated